src/Plugin/TagMappingType/MediaEntity.php

```diff
@@ -178,7 +181,7 @@ public function calculateDependencies() {
   /**
    * {@inheritdoc}
    */
-  public function replacePlaceholder(IDMS $idms, $fieldItem) {
+  public function replacePlaceholder(IdmsBuilderInterface $builder, IDMS $idms, $fieldItem) {
```

Comment:
The TagMappingType should not need any knowledge of builder details. I would not inject the builder here, as it is much too specific.

I'd rather go with something like $buildMode being either '_singlefile_' or '_multifile_' or with providing an additional interface for TagMappingTypes, that can provide multiple files.